### ISSUE TITLE
[9.3] Government Bank Account Issue

### ISSUE THREAD

**Issue by Pr0n3** [StrangeLoopGames/EcoIssues#21391]
As soon as we are using government bank accounts on the shops people can't sell items anymore,

- Tested it with peronal account **works**
- Tested it with a new bank account with managers with a government role **works**
- Tested it with a government account **is not working**

This error appeared for someone who tried to use a shop with the Treasury Account: https://i.imgur.com/EDYkf8d.jpg

**Comment by MentalMonk** [comment]
Probably a useless comment other than confirming Pr0n3's report to the T and posting a crash report.
[Crash Report.txt](https://github.com/StrangeLoopGames/EcoIssues/files/6319321/Crash.Report.txt)

**Comment by MentalMonk** [comment]
Store wont accept any trades while using a government account, but now it seems when you make a new government account it creates a personal gov account.
![image](https://user-images.githubusercontent.com/81957057/114900423-6edfeb80-9e46-11eb-94a4-e39835918179.png)

**Comment by IcedForge** [comment]
Seems like some permission calls breaking when trying to get funds from the government accounts.

Here is a recurring transfer crash 
[log_210415113125.log.txt](https://github.com/StrangeLoopGames/EcoIssues/files/6320251/log_210415113125.log.txt)

**Comment by IronyTroll** [comment]
Good Morning Team,

Can confirm, all our government stores no longer accept trades as they are using government bank accounts.

https://youtu.be/fHxOCTo4T0s

**Comment by Daniel-Magulado** [comment]
Tested on 0.9.3.1 Release-204.

The issue is still happening.

Repro:
1. Player 1 sets a store. Player 1 holds a Title 1 with access to a Government Bank account such as a treasury.
2. Player 1 sets Treasury as Bank Account for the Store.
3. Player 1 sets for a buying list on his/her store.
4. Player 1 resigns from Title 1.
5. Player still has his store set its bank account to Treasury (Government Bank Account)
6. Player 2 comes in and sells items to Player 1's store.
7. Clicking Trade, nothing happens.
8. If you buy and sell at the same time, it will work.

Video:
https://images.zenhubusercontent.com/47210472/bd37fa70-3185-484e-ac16-e7201763cd64/2021_04_17_03_05_trim.mp4

**Comment by Daniel-Magulado** [comment]
Additional Info from other user report: #21436

> On the server that I'm playing on the store work just fine except for the government store. When someone tries to sell something to a store using a government account, the trade button does nothing when pressed. The store has been replaced a few times, players have re-logged (server also restarted since the previous update) but still nothing. The currency of the store has also been changed to using a player currency (the one where the player has infinite if it's their currency) and still nothing. The store is on land that is own by the people who are in the government.

**Comment by IcedForge** [comment]
Yes thank you all for the super helpful reports and explanations.

The issue revolving this lies in when a title does  not contain a valid citizen so make sure someone holds the title to be able to perform transactions.
